storage/innobase/btr/btr0cur.cc

```diff
@@ -4402,6 +4431,19 @@ btr_cur_pessimistic_delete(
 	page = buf_block_get_frame(block);
 	index = btr_cur_get_index(cursor);
 
+#ifdef WITH_INNODB_SCN
+	if (innodb_use_scn && index->is_clust()) {
+		if (!index->online_log)
+		{
+			scn_mgr.batch_write(block, mtr);
+		}
+		else
+		{
+			block->clear_cursor();
+		}
+	}
+#endif
```

Comment:
This code is being duplicated in many places. I think that it would be better to write a separate function for it.